src/modbus.h

```diff
@@ -92,6 +94,7 @@ MODBUS_BEGIN_DECLS
 #define MODBUS_MAX_WRITE_REGISTERS         123
 #define MODBUS_MAX_WR_WRITE_REGISTERS      121
 #define MODBUS_MAX_WR_READ_REGISTERS       125
+#define MODBUS_MAX_REFERENCE_FILES 10
 
```

Comment:
NAK. This does not belongs to this group here, since it's a implementation specific maximum, not a maximum defined by the standard or derived from it. So it should stand alone...

Reply:
The standard says : 
While it is allowed for the File Number to be in the range 1 to 0xFFFF, it should be noted that
interoperability with legacy equipment may be compromised if the File Number is greater than
10 (0x0A).
So imho this fits there quite well and is not purely implementation specific. ( I renamed it to 
MODBUS_MAX_FILE_RECORDS )

Reply:
correction renamed it to: MODBUS_MAX_RECORD_FILES .
I also added the comment from the standard to the header, so one may see, where the number came from.

Reply:
We don't restrict registers to 10000, we use the full 16bit address space, there's absolutely no reason to imply that the protocol only allows 10 either.  I strongly agree with @mhei's comment here, this is not part of a standard to limit it to 10.

Reply:
I guess you got it wrong, the 10 is the number of files, not the number of records in each file. This might be up to 64k. The maximum number of records is controlled by the mapping-function. You can individually allocate your needed number per file. Up to 64k Records * 10 Files.

Reply:
stahp.  straight out of app protocol 1.1b. page 33
```
file number: 0x0001 to 0xFFFF
record number: 0x0000 to 0x270F
```

there are then footnotes that "comaptibility with legacy gear may be poor if you actually _use_ file numbers higher than 10"

using file numbers in the "old" "comaptible" range or higher numbers is a _user_ choice.

Reply:
I'm still not convinced, but i changed it to 0xffff.

Reply:
I have to remove the checks as well due to the limited range of uint16_t , which makes it less safe, when someone reduces the number :(